# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/rect.glsl
# Commit: a0b3e2669a2d
# Full Hash: a0b3e2669a2dc49c467521cb9b7966ca05e3dbc3
# Author: Ashley Hale <ahale@mozilla.com>
# Date: 2024-05-08 09:49:23
# Regressor Bug: 1824502
# File Overlap Count: 1
# Description:
#   Bug 1824502 - Implement native SVG filter rendering in WebRender r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209646
# ==============================================================================

diff -r d1df40ba7c88 -r a0b3e2669a2d gfx/wr/webrender/res/rect.glsl
--- a/gfx/wr/webrender/res/rect.glsl	Wed May 08 01:43:25 2024 +0000
+++ b/gfx/wr/webrender/res/rect.glsl	Wed May 08 04:38:26 2024 +0000
@@ -38,3 +38,12 @@
 vec2 rect_size(RectWithEndpoint rect) {
     return rect.p1 - rect.p0;
 }
+
+// this is similar to rect_clamp but repeats the image for coordinates outside
+// the rect, used in SVG feTile filter
+vec2 rect_repeat(vec2 p, vec2 p0, vec2 p1) {
+    vec2 r = p - p0;
+    vec2 s = p1 - p0;
+    vec2 is = 1.0 / max(s, vec2(0.000001));
+    return p0 + s * fract(is * r);
+}